h2o-algos/src/main/java/hex/schemas/GBMV3.java

```diff
@@ -98,6 +100,11 @@ public static final class GBMParametersV3 extends SharedTreeV3.SharedTreeParamet
     @API(help="A set of allowed column interactions.", level= API.Level.expert)
     public String[][] interaction_constraints;
 
+    @API(help="Create checkpoints into defined directory while training process is still running. In case of cluster shutdown, this checkpoint can be used to restart training.", level = API.Level.expert, gridable = false)
+    public String in_training_checkpoints_dir;
+
+    @API(help="Checkpoint the model after every so many trees. Parameter is used only when in_training_checkpoints_dir is defined", level = API.Level.expert, gridable = false)
+    public int in_training_checkpoints_tree_interval;
```

Comment:
as soon as we have the fields list above to filter in those used by GBM, shouldn't those attributes be defined in `SharedTreeV3`?

Reply:
I put them only in GBMV3.java intentionally, we want to support only GBM in first implementation.

Reply:
so you're saying that the `fields` attribute in that class isn't enough to ensure that those attributes (once moved to `SharedTreeV3`) are enabled only for GBM?

I think that's what we do though for example with `histogram_type` : this attribute is defined in `SharedTreeV3` (makes sense as it's mapped to `SharedTreeParameters`), and then it's activated in DRF and GBM simply by adding the name in the `fields` list. On the other side, it's not enabled in XGBoostV3 and so nothing is generated on the clients... looks cleaner to me as it keeps the 1-1 relationship between the Schema class and the Parameters class.
Apparently, same with `check_constant_response`

Reply:
Aaa understand, there is fields array for that purpose 🤦